vortex-layout/src/reader.rs

```diff
@@ -32,16 +33,13 @@ pub trait LayoutReader: 'static + Send + Sync {
 
     /// Returns the number of rows in the layout reader.
     /// An inexact count may be larger or smaller than the actual row count.
+    /// FIXME(ngates): remove this.
     fn row_count(&self) -> Precision<u64>;
 
-    /// Register the splits of this layout reader.
-    // TODO(ngates): this is a temporary API until we make layout readers stream based.
-    fn register_splits(
-        &self,
-        field_mask: &[FieldMask],
-        row_offset: u64,
-        splits: &mut BTreeSet<u64>,
-    ) -> VortexResult<()>;
+    /// Given a [`RowSelectionRef`] which can answer range included queries.
```

Comment:
This comment doesn't quite make sense. 

Also, you should explain that the returned masks don't need to be an exact intersection with the row selection. Although I guess the trait doesn't allow you to perform one anyway.